### PR TITLE
feat(consents): registration for FIDO flow

### PR DESCRIPTION
**Pull request by eoln**
- in pisp documentation is standing simple sentence that `The PublicKeyCredential object - which contains the key's Id, and an AuthenticatorAttestationResponse which contains the public key` https://github.com/mojaloop/pisp/blob/master/docs/linking/README.md#162-registering-the-credential

- but it turns out to be a very complex type structure (comparing FIDO to GENERIC flow) with a great impact on auth-service registration and validation code

This PR contains changes in interfaces needed to deliver FIDO related implementation in auth service

### CODE REVIEW

File: tests/dto/thirdparty.test.ts [hunk 1]
```diff
@@ -101,22 +101,34 @@ describe('thirdparty', () => {
...
+  const fidoPublicKeyCredential: Schemas.FIDOPublicKeyCredential = {
+    id: 'some-credential-id',
     response: {
-      clientDataJSON: 'client-data'
+      clientData: {
+        challenge: 'the-challenge',
+        origin: 'pisp.mojaloop.io',
+        type: 'webauthn.create'
+      },
+      attestation: {
```
Comment (kleyow) [inline tests/dto/thirdparty.test.ts:112]: I totally overlooked the `attestationObject` :sigh:

File: thirdparty/openapi3/components/schemas/ConsentsPostRequest.yaml [hunk 2]
```diff
@@ -15,12 +15,12 @@ properties:
     description: |
       The id of the ConsentRequest that was used to initiate the
       creation of this Consent.
-  credential:
-    $ref: ./SignedCredential.yaml
+
```
Comment (kleyow) [inline thirdparty/openapi3/components/schemas/ConsentsPostRequest.yaml:18]: Why is this removed? I need it for the initial POST consents
image

File: thirdparty/openapi3/components/schemas/ConsentsPostRequest.yaml [hunk 3]
```diff
@@ -15,12 +15,12 @@ properties:
...
-    $ref: ./SignedCredential.yaml
+  
   scopes:
     type: array
     items:
       $ref: ./Scope.yaml
 required:
   - consentId
   - scopes
+  - consentRequestId  # we need to obligatory store the reference to request process
```
Comment (kleyow) [inline thirdparty/openapi3/components/schemas/ConsentsPostRequest.yaml:26]: I guess that it must of been a error in the docs since I saw it missing from the request
image

File: thirdparty/openapi3/components/schemas/ConsentsPostRequest.yaml (hunk 2 above)
Comment (eoln) [reply thirdparty/openapi3/components/schemas/ConsentsPostRequest.yaml:18]: we have two POST /consents
- the first when DFSP asks PISP to deliver the credential: https://github.com/mojaloop/pisp/blob/master/docs/linking/README.md#15-grant-consent
- the second when DFSP ask AUTH-SERVICE to register the consent with credential (as action on PUT /consent/{ID} call made by PISP to DFSP
https://github.com/mojaloop/pisp/blob/master/docs/linking/README.md#162-registering-the-credential

in the first one, there is no need for credential prop
in the second, the credential property is mandatory

I think that we need two independent schemas for these two flows, elsewhere there will be a need to implement additional validation logic to check presence/absence of credential field.

Comment (eoln) [reply thirdparty/openapi3/components/schemas/ConsentsPostRequest.yaml:18]: but defining two independent schemas for the same path and method would be a challenge. 
Which schema should be used is strictly dependent on the receiver context - PISP or AUTH-SERVICE

Comment (kleyow) [reply thirdparty/openapi3/components/schemas/ConsentsPostRequest.yaml:18]: > I think that we need two independent schemas for these two flows, elsewhere there will be a need to implement additional validation logic to check presence/absence of credential field.

Sounds good. I like the multiple independent schemas better.

File: thirdparty/openapi3/components/schemas/ConsentsPostRequest.yaml (hunk 3 above)
Comment (eoln) [reply thirdparty/openapi3/components/schemas/ConsentsPostRequest.yaml:26]: `consentRequestId` is mandatory in Grant Consent phase when DFSP ask PISP to deliver credential
https://github.com/mojaloop/pisp/blob/master/docs/linking/README.md#15-grant-consent and `credential` not

**Review by kleyow — DISMISSED** [review]
Looks good.

**Comment by kleyow** [comment]
Whoops. Tests are needed.

**Comment by eoln** [comment]
@kleyow tests fixed

**Review by kleyow — APPROVED** [review]
